src/scripts/sniperino.ts

```diff
@@ -0,0 +1,139 @@
+// Description:
+//  Beat the sniperino
+//  
+//
+// Commands:
+//  sniperino( me)? - start a new game of sniperino 
+//  snipe( me)? - roll a snipe roll in an attempt to beat the sniperino
+//
+// Author:
+//  Steve Shipsey
+
+import { Robot } from "tsbot";
+
+interface ISniper {
+    name: string;
+    gamesPlayed: number;
+    gamesWon: number;
+    currentSnipe?: number;
+
+    getWinRate(): number;
+    generateNewSnipe();
+}
+
+class Sniper implements ISniper {
+    public name: string;
+    public gamesPlayed: number;
+    public gamesWon: number;
+    public currentSnipe: number;
+    
+    constructor(n: string) {
+        this.name = n.slice(0, n.indexOf(" "));
+        this.gamesPlayed = 0;
+        this.gamesWon = 0;
+    }
+
+    public getWinRate(): number {
+        // Calculation with some javascript rounding magic
+        //return this.gamesPlayed > 0 ? Number(Math.round((this.gamesWon / this.gamesPlayed * 100) + 'e2') + 'e-2') : 0;
+        return this.gamesPlayed > 0 ? Math.round(this.gamesWon / this.gamesPlayed * 100 * Math.pow(10, 2)) / Math.pow(10,2) : 0;
+    }
+
+    public generateNewSnipe() {
+        this.currentSnipe = Math.floor(Math.random() * 99) + 1;
+    }
+}
+
+let game = (robot: Robot) => {
+    
+    // If the brain has no knowledge of sniperino, create it
+    if (robot.brain.get("sniperino") == null) {
+        robot.brain.set("sniperino", {});
+    }
+
+    // Get a sniper with a given name. If one doesn't exist, create it.
+    let getOrCreateSniper = (name: string) => {
+    
+        // Check for active sniper, if not, create one
+        if (robot.brain.get("sniperino")[name] == null) {
+            let s = robot.brain.get("sniperino");
+            s[name] = new Sniper(name);
+            robot.brain.set("sniperino", s);
+        }
+        return robot.brain.get("sniperino")[name];
+    }
+
+    // Just get a sniper. Return null on failure.
+    let getSniper = (name: string) => {
+        return robot.brain.get("sniperino")[name];
+    }
+
+    // Respond to a stats message
+    robot.respond(/stats( me)?/i, (res) => {
+
+        let snipers = robot.brain.get("sniperino");
+        let retStr = "";
+
+        // Create a stats dictionary
+        for (var s in snipers) {
+            retStr += `${s} : ${snipers[s].getWinRate()}%\n`;
```

Comment:
Don't do this. String append is slow in JS. Look at how I output stuff in the ilvl script